### DISCUSSION TITLE
Multi Domain Support not Cross Domain

### DISCUSSION THREAD

**Discussion by masterbater** [ory/kratos#2451] [Q&A] [answered]
Does this support multiple domains that proxy only one instance of ory kratos.
Example
foo.com and bar.com reverse proxy same kratos endpoint

Does the config only supports one domain for  redirect? Is it possible for redirect to guess based on the domain it is being accessed? Or the intended behavior is that only the main domain set like foo.com and using bar.com will always redirect to foo.com during browser flows?

**Comment by vinckr** [comment, marked as answer]
See this document for more info: https://www.ory.sh/docs/kratos/guides/multi-tenancy-multitenant.
You basically host one Ory Kratos instance for each tentant (domain). You can also build a more complex system using OAuth2 flows for federated identities (with [Ory Hydra](https://github.com/ory/hydra)) if you have many tenants (>~1000s)

We also recommend Ory Network to easily manage multiple tenants: https://www.ory.sh/docs/guides/console/create-project

**Reply by dm17** [reply]
If we're trying to do SSO in order to support multiple domains, then I'm not sure how that is supposed to look. It would be helpful if that multitenant doc were less sparse - including exactly how Ory defines multitenancy in this case. For example, if it is one FQDN per instance of Kratos, then which aspect of Ory is in charge of the SSO between those FQDNs? Will this SSO solution require domain forwarding? I'm not sure where Hydra fits in except for the fact that we need OAuth for one of the domains we need to enable SSO on.

**Reply by vinckr** [reply]
See here for an explanation for how multiple domains pointing to one Ory instance is solved in Ory Network: https://www.ory.com/docs/kratos/bring-your-own-ui/configure-ory-to-use-your-ui#multiple-custom-domains

**Comment by masterbater** [comment]
Thanks I also found a solution to just have one instance of kratos then
just use some reverse proxy to be available to multiple domain. Im waiting
for adding user attributes so I can add tenantId

**Reply by vinckr** [reply]
You mean metadata? https://www.ory.sh/docs/kratos/manage-identities/managing-users-identities-metadata

**Reply by omeraplak** [reply]
Hey @masterbater ,
We are also looking for a similar solution. Could you share some details about the solution you have found?
